File: old/dig1/dig/dig2/programs/nla/freire1.c
#include <stdio.h>
#include <assert.h>
#include <stdlib.h>  //required for afloat to work

int mainQ(float a){

  float x = x=a/2.0;
  int r = 0;
  while(1){
    //%%%traces: float a, float x, int r
    if (!(x>r)) break;
    x = x - r; r = r + 1;
  }

  //assert(r==(int)round(sqrt(a)));
  return r;
}


int main(int argc, char **argv){
  mainQ(atof(argv[1]));
  return 0;
}

File: examples/SymInferASE/nla/Freire1.java
public class Freire1 {

     public static void main (String[] args) {
	  mainQ(Integer.parseInt(args[0]));
     }

     public static int mainQ(int a) {
	  double x = ((double)a)/2.0;
	  int r = 0;


	  while(true){
	       //assert((double)a == 2*x + r*r - r); 
	       //%%%traces: int a, double x, int r
	  
	       if (!(x>r)) break;
	       x = x - r;
	       r = r + 1;
	  }

	  //assert(r==(int)round(sqrt(a)));
	  return r;	  

     }
}
